rich/console.py

```diff
@@ -1947,7 +1947,7 @@ def _caller_frame_info(
         frame = currentframe()
         if frame is not None:
             # Use the faster currentframe where implemented
-            while offset and frame:
+            while offset and frame is not None:
```

Comment:
frame doesn't have `__bool__` or `__len__`, so it will always evaluate to true. This makes it explicit.